File: lynnedslag/lynnedslag_linear.py
n, m = map(int, input().split())

a = [False] * n
a[0] = True

for _ in range(m):
    l, r = map(int, input().split())
    r += 1
    a[l] = not a[l]
    if r < n:
        a[r] = not a[r]

total = 0
cur = False
for x in a:
    if x:
        cur = not cur
    total += cur

print(total)

File: lynnedslag/lynnedslag_linear.cpp
#include <iostream>
#include <vector>

int main() {
  std::cin.tie(nullptr);
  std::ios_base::sync_with_stdio(false);

  int n, k;
  std::cin >> n >> k;

  std::vector<bool> a(n, false);
  a[0] = true;

  for (int i = 0; i < k; i++) {
    int l, r;
    std::cin >> l >> r;
    r += 1;

    a[l] = !a[l];
    if (r < n)
      a[r] = !a[r];
  }

  int total = 0;
  bool cur = false;
  for (bool x : a) {
    if (x) {
      cur = !cur;
    }
    total += cur;
  }

  std::cout << total;
}
